profiler/hotapp/main.go

```diff
@@ -75,6 +75,18 @@ func allocImpl() {
 	}
 }
 
+// Simulates a function which allocates a lot of memory, but does not hold on
+// to that memory.
+func allocMany() {
+	// Allocate 1 MiB of 64 KiB chunks repeatedly.
+	for {
+		for i := 0; i < 16; i++ {
+			_ = make([]byte, 64*1024)
+		}
+		time.Sleep(time.Second)
```

Comment:
This loop allocates 1 MiB per second which isn't a lot and that's 10 MiBs per the profile duration of 10 seconds. The variation between profiles may be significant since the sampling threshold is 512 KiB by default. And I wouldn't be surprised if some profiles will miss samples altogether. I would increase the allocation rate at least 10x, maybe 100x. Basically, I would increase it sufficiently to make the allocation metric consistent enough between individual profiles (you can look at it using the weight filter which will show the distribution), but also not too much to avoid too much CPU time spent in GC.

Reply:
This now allocates 100MiB per second.
Over 71 profiles, allocMany has an average of 104.7MiB. Weight filters show profiles have between 50MiB and 140MiB.

Reply:
I don't think it's 100 MiB per second, I think it's per 10 seconds? Should be fine if at least average works out to be ~100 MiB per 10 second profile, just confirming the math.

Reply:
Looks good, I'll let the repo owners to approve the PR now.